software/firmware/europi.py

```diff
@@ -467,29 +468,60 @@ def __init__(
         i2c = I2C(channel, sda=Pin(sda), scl=Pin(scl), freq=freq)
         self.width = width
         self.height = height
-
+        self.writers = {}  # re-usable large font writer instances
         if len(i2c.scan()) == 0:
             if not TEST_ENV:
                 raise Exception(
                     "EuroPi Hardware Error:\nMake sure the OLED display is connected correctly"
                 )
         super().__init__(self.width, self.height, i2c)
 
-    def centre_text(self, text):
-        """Split the provided text across 3 lines of display."""
+    def writer(self, font):
+        """Returns the large font writer for the specified font."""
+        n = font.__name__
+        if n not in self.writers:
+            self.writers[n] = Writer(self, font)
+        return self.writers[n]
+
+    def text(self, s, x, y, c=None, font=None):

```

Comment:
Why is here a c parameter again? It's not used later in the function?
-edit:
super().text() does accept a c (color) parameter: FrameBuffer.text(s, x, y[, c])